projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -986,12 +1017,22 @@ export class IgxTimePickerComponent implements
 
     private _getSelectedTime(): Date {
         const date = this.value ? new Date(this.value) : new Date();
-        date.setHours(parseInt(this.selectedHour, 10));
-        date.setMinutes(parseInt(this.selectedMinute, 10));
+        if (this.selectedHour) {
+            date.setHours(parseInt(this.selectedHour, 10));
+        }
+        if (this.selectedMinute) {
+            date.setMinutes(parseInt(this.selectedMinute, 10));
+        }
         date.setSeconds(0);
-        if (this.selectedAmPm === 'PM' && this.selectedHour !== '12') {
+        if (this.showHoursList && this.selectedAmPm === 'PM' && this.selectedHour !== '12') {
```

Comment:
merge conditions with the below line if possible to avoid duplicating code (date.setHours(date.getHours() + 12))

Reply:
I'd rather not, this logic is very fragile.

Reply:
You could merge those with `||` and the result will be the same.